### PR TITLE
Switch from gometalinter to golangci-lint

### PR DESCRIPTION
**Pull request by gpaul**
The golangci-lint linter is less resource intensive than gometalinter. Also, gometalinter has been deprecated in favour of golangci-lint.

Hopefully this fixes a rare linting timeout when the CI server is busy.

Fixes https://jira.mesosphere.com/browse/DCOS_OSS-2415

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -37,12 +37,10 @@ ENV GOPATH /go
...
 
-RUN mkdir -p /go/src/github.com/alecthomas && \
-    cd /go/src/github.com/alecthomas && \
-    git clone https://github.com/alecthomas/gometalinter.git --branch=v1.2.1 && \
-    go install -v github.com/alecthomas/gometalinter && \
-    gometalinter --install && \
-    go get -u golang.org/x/tools/cmd/goimports && \
+RUN mkdir -p /go/src/github.com/golangci/ && \
+    cd /go/src/github.com/golangci && \
+    git clone https://github.com/golangci/golangci-lint.git --branch=v1.17.1 && \
```
Comment (bbannier) [inline Dockerfile]: Let's move the version number into a constant like we do for other dependencies for easier maintenance.

File: pkg/csilvm/csilvm_test.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // +build !unit
 
+//nolint:errcheck
```
Comment (bbannier) [inline pkg/csilvm/csilvm_test.go]: I feel we should also add a `TODO` here ;)

File: pkg/lvm/lvm_test.go [hunk 3]
```diff
@@ -1,5 +1,6 @@
 // +build !unit
 
+//nolint:errcheck
```
Comment (bbannier) [inline pkg/lvm/lvm_test.go]: `TODO`?

File: pkg/lvm/lvm.go [hunk 4]
```diff
@@ -859,8 +859,14 @@ func run(cmd string, v interface{}, extraArgs ...string) error {
 		// We use Lock instead of TryLock as we have no alternative way of
 		// making progress. We expect lvm2 command-line utilities invoked by
 		// this package to return within a reasonable amount of time.
-		lvmlock.Lock()
-		defer lvmlock.Unlock()
+		if lerr := lvmlock.Lock(); lerr != nil {
+			return lerr
```
Comment (bbannier) [inline pkg/lvm/lvm.go]: Looking at the impl of flock the only possible errors we might get would be related to failures using the configured file. Does it make sense to decorate this error with that additional context so users see e.g., `Could not lock configured file lock: ...`?

**Comment by gpaul** [comment]
@nfnt @bbannier I've addressed all comments. I've introduced  two functions: `check` which panics when `err != nil` and `try`, which just logs the error. The `try` function is necessary because if a test reaches the greater test cleanup function, it performs its own cleanup but we also want to defer cleanup of resources as we create them, in case a subsequent step fails before the greater test cleanup is reached and scheduled.

**Comment by gpaul** [comment]
I'll squash the commits once the PR is approved.

File: pkg/csilvm/csilvm_test.go (hunk 2 above)
Comment (bbannier) [reply pkg/csilvm/csilvm_test.go]: I like `try` and `check` much more than a `TODO`.

**Pull request merged**